packages/ckeditor5-table/src/tablecolumnresize/tablecolumnresizeediting.js

```diff
@@ -484,88 +388,59 @@ export default class TableColumnResizeEditing extends Plugin {
 		domEventData.preventDefault();
 		eventInfo.stop();
 
-		this._isResizingActive = true;
-		this._resizingData = this._getResizingData( domEventData );
+		const target = domEventData.target;
+		const modelTable = editor.editing.mapper.toModelElement( target.findAncestor( 'figure' ) );
+		const viewTable = target.findAncestor( 'table' );
 
-		editingView.change( writer => {
-			writer.addClass( 'table-column-resizer__active', this._resizingData.elements.viewResizer );
-		} );
-	}
+		// Calculate the dom column widths. It is done by taking the width of the widest cell
+		// from each table column (we relay on the TableWalker in terms of determining
+		// to which column the cell belongs).
+		const columnWidthsInPx = Array( getNumberOfColumn( modelTable, editor ) );
+		const tableWalker = new TableWalker( modelTable );
 
-	/**
-	 * Handles the `mouseup` event if previously the `mousedown` event was triggered from the column resizer element.
-	 *
-	 * @private
-	 * @param {module:utils/eventinfo~EventInfo} eventInfo
-	 * @param {module:engine/view/observer/domeventdata~DomEventData} domEventData
-	 */
-	_onMouseUpHandler() {
-		const editor = this.editor;
-		const editingView = editor.editing.view;
-
-		if ( !this._isResizingActive ) {
-			return;
-		}
-
-		const {
-			modelTable,
-			viewColgroup,
-			viewFigure,
-			viewResizer
-		} = this._resizingData.elements;
-
-		const columnWidthsAttributeOld = modelTable.getAttribute( 'columnWidths' );
-		const columnWidthsAttributeNew = [ ...viewColgroup.getChildren() ]
-			.map( viewCol => viewCol.getStyle( 'width' ) )
-			.join( ',' );
+		for ( const cellSlot of tableWalker ) {
+			const viewCell = editor.editing.mapper.toViewElement( cellSlot.cell );
+			const domCell = editor.editing.view.domConverter.mapViewToDom( viewCell );
+			const domCellWidth = getDomCellOuterWidth( domCell );
 
-		const isColumnWidthsAttributeChanged = columnWidthsAttributeOld !== columnWidthsAttributeNew;
+			if ( !this._columnIndexMap.has( cellSlot.cell ) ) {
+				this._columnIndexMap.set( cellSlot.cell, cellSlot.column );
+			}
 
-		const tableWidthAttributeOld = modelTable.getAttribute( 'tableWidth' );
-		const tableWidthAttributeNew = viewFigure.getStyle( 'width' );
+			if ( !columnWidthsInPx[ cellSlot.column ] || domCellWidth < columnWidthsInPx[ cellSlot.column ] ) {
+				columnWidthsInPx[ cellSlot.column ] = toPrecision( domCellWidth );
+			}
+		}
 
-		const isTableWidthAttributeChanged = tableWidthAttributeOld !== tableWidthAttributeNew;
+		// Insert colgroup for the table that is resized for the first time.
+		if ( ![ ...domEventData.target.findAncestor( 'table' ).getChildren() ].find( viewCol => viewCol.is( 'element', 'colgroup' ) ) ) {
+			editingView.change( viewWriter => {
```

Comment:
I'd look to extract it into a separate helper function that would be just responsible for adding a `colgroup` to the `table`.

It might require a lot of parameters though, so feel free to check it and discard just in case.